substrate/frame/distribution/src/functions.rs

```diff
@@ -0,0 +1,149 @@
+pub use super::*;
+impl<T: Config> Pallet<T> {
+
+
+	pub fn pot_account() -> T::AccountId{
+		// Get Pot account
+		let pot_id = T::PotId::get();
+		let pot_account: T::AccountId = pot_id.into_account_truncating();
+		pot_account
+	}
+
+	pub fn get_spending(project_account: T::AccountId) -> Vec<SpendingIndex> {
+		let mut spendings: Vec<SpendingIndex>  = Vec::new();		
+		let mut value = Some(project_account);
+		
+		for spending in Spendings::<T>::iter() {
+			let info = spending.1;
+			if info.whitelisted_project == value {
+				spendings.push(spending.0);
+			} 
+		} 
+
+		spendings
+	}
+
+    /// Series of checks on the Pot, to ensure that we have enough funds
+	/// before executing a spending
+	pub fn pot_check(amount: BalanceOf<T>) -> DispatchResult {
+		
+		// Get Pot account		
+		let pot_account: T::AccountId = Self::pot_account();
+
+		// Check that the Pot as enough funds for the transfer
```

Comment:
I'm not sure why you need to check the project account has balance? The project having enough balance is only relevant when they are claiming their portion

Reply:
I am actually using this check before creating a new spending & locking the necessary funds (cannot lock what we don't have...)